pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -71,6 +71,12 @@ def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, use_connect_ar
 
     if isinstance(syn_spec, str):
         return {"synapse_model": syn_spec}
+
+    if (not isinstance(syn_spec, CollocatedSynapses) and
+        "synapse_model" in syn_spec and
+            not isinstance(syn_spec["synapse_model"], str)):
+        raise kernel.NESTError("'synapse_model' must be a string or a CollocatedSynapses object")
+
     elif isinstance(syn_spec, dict):
```

Comment:
Inserting another ``if`` before this ``elif`` changes the logic of when the ``else`` part will be triggered. I don't think we need the ``else`` part here anyway, as the previous ``if``s either return or raise an exception. Could you change this ``elif`` into an ``if``, and move the two lines of code from below:
```python
elif isinstance(syn_spec, CollocatedSynapses):
  return syn_spec
```
to the top of the function?